cmd/processor/app/processor.go

```diff
@@ -239,7 +247,12 @@ func (p *Processor) GetStatus() status.Status {
 
 	// if any worker isn't ready yet, return initializing
 	for _, workerInstance := range workers {
-		if workerInstance.GetStatus() != status.Ready {
+		switch workerInstance.GetStatus() {
+		case status.Error:
+			return status.Error
+		case status.Ready:
+			continue
```

Comment:
what's the plan here? waiting until the worker is in error/intializating? perhaps you mean "break"?